org.adempiere.base/src/org/compiere/model/PO.java

```diff
@@ -1343,6 +1375,26 @@ protected void load (int ID, String trxName)
 		}
 	}	//	load
 
+	/**
+	 * Load record with UUID
+	 * 
+	 * @param uuID    UUID
+	 * @param trxName transaction name
+	 */
+	protected void loadByUU(String uuID, String trxName)
+	{
+		// reset new values
+		m_newValues = new Object[get_ColumnCount()];
+		m_uuid = uuID;
+		checkImmutable();
+
+		if (log.isLoggable(Level.FINEST))
+			log.finest("uuID=" + uuID);
+		if (!Util.isEmpty(m_uuid, true))
+		{
+			load(trxName);
+		}
+	} // loadByUU
```

Comment:
if the definition here is if uuid==null, it should create a new PO record, it should be simplify to just if (Util.isEmpty(uuid, true)) load(0, trxName); else load(uuid, trxName);

Reply:
This is done in latest commit